### PR TITLE
slices: implement SliceRandom()

### PR DESCRIPTION
**Pull request by karasz**

### CODE REVIEW

File: slices.go [hunk 1]
```diff
@@ -150,3 +157,20 @@ func SliceCopyFn[T any](s []T,
...
+func SliceRandom[T any](a []T) T {
+	var result T
+
+	switch len(a) {
+	case 0:
+		return result
+	case 1:
+		result = a[0]
+	default:
+		rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
```
Comment (amery) [inline slices.go:171]: shouldn't we _assume_ someone already initialised `rand.Reader` before calling `SliceRandom()`?

Comment (amery) [reply slices.go:171]: we do on `acmefy/pkg/ca` ....

File: slices.go [hunk 2]
```diff
@@ -150,3 +157,20 @@ func SliceCopyFn[T any](s []T,
 
 	return result
 }
+
+// SliceRandom returns a random element from a slice
+func SliceRandom[T any](a []T) T {
```
Comment (amery) [inline slices.go:162]: returning `(T, bool)` maybe?

File: slices.go (hunk 1 above)
Comment (amery) [reply slices.go:171]: We could be calling this function a lot, and initialising a new source every this time sounds... excessive

Comment (karasz) [reply slices.go:171]: that might be true. Let me think about it.

**Review by amery — APPROVED** [review]

File: slices_test.go [hunk 3]
```diff
@@ -7,12 +7,15 @@ import (
 	"golang.org/x/exp/slices"
 )
 
+// revive:disable
```
[the remark this answers is not in the document]
Comment (karasz) [reply slices_test.go:10]: testing should not have revive, but there is no switch for that.... yet

[the remark this answers is not in the document]
Comment (amery) [reply slices_test.go:10]: I don't agree with that statement

**Comment by amery** [comment]
`v0.9.2` tagged

**Pull request merged**